packages/js-lib/src/ajv/ajv.validations.test.ts

```diff
@@ -0,0 +1,989 @@
+/* eslint-disable vitest/valid-expect */
+/* eslint-disable id-denylist */
+// oxlint-disable no-unused-expressions
+
+import { describe, expect, test } from 'vitest'
+import { localDate } from '../datetime/localDate.js'
+import { localTime } from '../datetime/localTime.js'
+import { j2 } from '../json-schema/jsonSchemaBuilder2.js'
+import { Set2 } from '../object/set2.js'
+import type { Branded, IsoDate, IsoDateTime } from '../types.js'
+import { AjvSchema } from './ajvSchema.js'
+
+describe('string', () => {
+  test('should work correctly with type inference', () => {
+    const schema = j2.string()
+
+    const [err, result] = AjvSchema.create(schema).getValidationResult('foo')
+
+    expect(err).toBeNull()
+    expect(result).toBe('foo')
+    result satisfies string
```

Comment:
Idea how to not make linters scream at this line. What if we combine it with other line? Like:


```suggestion
    expect(result satisfies string).toBe('foo')
```